sdks/python/apache_beam/runners/interactive/interactive_environment.py

```diff
@@ -501,9 +501,11 @@ def evict_computed_pcollections(self, pipeline=None):
     is specified, evicts for all pipelines.
     """
     if pipeline:
+      discarded = set()
       for pcoll in self._computed_pcolls:
         if pcoll.pipeline is pipeline:
-          self._computed_pcolls.discard(pcoll)
+          discarded.add(pcoll)
+      self._computed_pcolls -= discarded
```

Comment:
Don't you need to call discard on `_computed_pcolls`?

Reply:
I perform the set subtraction on L508 which takes care of it.

Reply:
ah makes sense